exporter/datadogexporter/factory.go

```diff
@@ -154,7 +154,8 @@ func createMetricsExporter(
 		cfg,
 		set,
 		pushMetricsFn,
-		exporterhelper.WithTimeout(cfg.TimeoutSettings),
+		// explicitly disable since we rely on http.Client timeout logic.
```

Comment:
Will this be confusing if some exporters use this `TimeoutSettings` as the timeout setting for the entire operation where as other exporters use it per network call? This makes me think we should have a different configuration option for it. What do you think?

Reply:
I have tried to use a pattern that is already used in the Collector in other exporters:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/243d6423992a4a76db30eb254d715ccb3a23c9e4/exporter/signalfxexporter/factory.go#L121-L122 https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/243d6423992a4a76db30eb254d715ccb3a23c9e4/exporter/lokiexporter/factory.go#L71-L72 https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/243d6423992a4a76db30eb254d715ccb3a23c9e4/exporter/splunkhecexporter/factory.go#L105-L106 https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/243d6423992a4a76db30eb254d715ccb3a23c9e4/exporter/observiqexporter/exporter.go#L42-L43

I would expect most users not to care about a global `Consume[Metrics/Traces/Logs]` function timeout (they don't even need to know that such a function exists to use the exporter). If we use different options I think we should have a wider conversation to have a consistent solution for all exporters that do this differently today.

Reply:
Agreed, I'm happy to approve this PR. It would be good to have that discussion if the pattern of disabling this timeout already happening, maybe the global timeout is less important.

Reply:
Opened open-telemetry/opentelemetry-collector#4497 to discuss this